### PR TITLE
Use string for model name in npred_signal()

### LINKED ISSUE #3386: `Dataset.npred_signal()` should take model name
`Dataset.npred_signal()` takes a model as argument, whereas in most other places (eg:`FluxPointEstimator` we access models by their names. This create some confusion.

@adonath shall I put in a PR to change this?

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR fixes #3386

### CODE REVIEW

**Comment by AtreyeeS** [comment]
I imagine that the doc test fail is unrelated.

**Comment by adonath** [comment]
Thanks @AtreyeeS, please remove the unrelated `test_stack.ipynb` notebook.

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -407,10 +407,13 @@ def npred_signal(self, model=None):
         """
         npred_total = Map.from_geom(self._geom, dtype=float)
 
-        for evaluator in self.evaluators.values():
-            if model is evaluator.model:
-                return evaluator.compute_npred()
+        if model is not None:
+            try:
+                return self.evaluators[model].compute_npred()
+            except:
```
Comment (adonath) [inline gammapy/datasets/map.py:413]: It's important to always specify the kind of error to catch like `except KeyError`. Also the new error message you give is not really an improvement (it missed which key is not contained). My proposal would be to just keep the `KeyError`, that is raised if the `model_name` does  not exist.

File: gammapy/datasets/map.py [hunk 2]
```diff
@@ -395,8 +395,8 @@ def npred_signal(self, model=None):
 
         Parameters
         -------------
-        model: `~gammapy.modeling.models.SkyModel`, optional
-            Sky model to compute the npred for.
+        model: str
```
Comment (adonath) [inline gammapy/datasets/map.py:398]: Maybe rename to `model_name` to make the change more explicit.

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @AtreyeeS, I have left two minor inline comments...

**Comment by adonath** [comment]
Thanks @AtreyeeS, once the remaining test fails are solved this is real to merge...

**Review by adonath — APPROVED** [review]
Thanks @AtreyeeS! Remaining test fails are unrelated...

**Pull request merged**